cpp/include/cudf/aggregation.hpp

```diff
@@ -90,6 +94,10 @@ class aggregation {
   }
 
   virtual ~aggregation() = default;
+
+  // override functions for compound aggregations
+  virtual std::vector<aggregation::Kind> get_simple_aggregations(data_type col_type) const;
```

Comment:
These are only used by groupby, yes? Therefore I don't think they should be part of the `aggregation` base class.

Reply:
Right now, yes. These could be extended to sort based groupby, and rolling too.

Reply:
Okay, sure. I like how this design is shaping up.